gradualizer

```diff
@@ -1,10 +1,26 @@
--module(gradualizer_cli).
--export([main/1]).
+#!/usr/bin/env escript
+%% -*- erlang -*-
+%%!
+
+-define(ebin_dir, "_build/default/lib/gradualizer/ebin").
```

Comment:
can we remind me of the purpose of the cli script? (apart from running it from a standalone git checkout)

I am a bit concerned hardcoding a rebar3 specific `ebin_dir` here (and also bit concerned that the escript won't be self-contained any more). I'm not sure this will work as a mix or erlang.mk dep (but there could be separate plugins for these tools). Also I guess this won't work if gradualizer is a rebar3 dependency (but there is rebar3 plugin for that case already)

I wonder if there are any other use-cases where it's the cli that should be used but it won't work?

Reply:
I see the cli just as another way of interacting with Gradualizer. Its purpose is not to specifically to help with the interaction towards mix (which there is already a plugin in the works for) or erlang.mk but it should be possible to use the cli there as with any cli. We require rebar3 to build so those who want to build the cli will need to use rebar3 and we should be fine with that. Hardcoding is unfortunate but there is no good way around it. In light of https://github.com/erlang/rebar3/issues/1936 I believe that what @zuiderkwast has presented here is the correct solution.

Reply:
I see, so the cli is not just that single escript but the repo as a whole. And the supported way of using the cli is to: clone the repo+`make app`+(optionally add the repo dir to the PATH).
I agree with you (myself as an ignorant example :) ) to highlight usage in the README. (eg it is not supported to copy or symlink just the escript file somewhere else - `filename:dirname(?FILE)` would give wrong output)

Reply:
While it would be nice that the CLI is self contained, it's OK if it's not (e.g. all node cli's are shipped this way). As you say, good documentation regarding this is key. Let's revisit this topic once OTP/rebar3 ships with standardized support for `/priv` inside escripts.